implants/lib/eldritch/src/pivot/arp_scan_impl.rs

```diff
@@ -1,5 +1,296 @@
 use anyhow::Result;
+use ipnetwork::{IpNetwork, Ipv4Network};
+#[cfg(not(target_os = "windows"))]
+use pnet::{
+    datalink::{self, channel, Channel::Ethernet, NetworkInterface},
+    packet::{
+        arp::{ArpOperations, ArpPacket, MutableArpPacket},
+        ethernet::{EtherType, EthernetPacket, MutableEthernetPacket},
+        Packet,
+    },
+    util::MacAddr,
+};
+use starlark::collections::SmallMap;
+use starlark::const_frozen_string;
+use starlark::values::{dict::Dict, Heap};
+use std::collections::HashMap;
+use std::net::{IpAddr, Ipv4Addr};
+use std::str::FromStr;
+use std::sync::{Arc, Mutex};
+use std::time::{Duration, SystemTime};
 
-pub fn arp_scan(_target_cidrs: Vec<String>) -> Result<Vec<String>> {
-    unimplemented!("Method unimplemented")
-}
\ No newline at end of file
+#[cfg(not(target_os = "windows"))]
+#[derive(Debug, Clone, PartialEq)]
+pub struct ArpResponse {
+    _source_ip: Ipv4Addr,
+    source_mac: MacAddr,
+    interface: String,
+}
+
+#[cfg(not(target_os = "windows"))]
+fn start_listener(
+    interface: NetworkInterface,
+    data: Arc<Mutex<HashMap<Ipv4Addr, Option<ArpResponse>>>>,
+) {
+    if interface.ips.iter().filter(|ip| ip.is_ipv4()).count() == 0 {
+        return;
```

Comment:
Would it make sense to return an error? Might be nice in case something breaks to know why.